src/classes/trackers.js

```diff
@@ -452,18 +462,38 @@ class Trackers {
         }
     }
 
+    /**
+     * Returns false if the given rule has an unsupported rule action, true
+     * otherwise.
+     * @param {TrackerRule} ruleObj
+     * @param {Set<string>} [supportedCustomRuleActions]
+     * @returns {boolean}
+     */
+    ruleActionSupported ({ action }, supportedCustomRuleActions) {
+        return (
+            // Rule action generally defaults to 'block' if omitted.
+            !action ||
+            // Standard rule actions are always supported.
+            Trackers.standardRuleActions.has(action) ||
+            // Provided custom rule actions (if any) are also supported.
+            (!!supportedCustomRuleActions && supportedCustomRuleActions.has(action))
+        )
+    }
+
     /**
      * Iterate through a tracker rule list and return the first matching rule, if any.
      * @param {TrackerObj} tracker
      * @param {RequestData} requestData
+     * @param {Set<string>} [supportedCustomRuleActions]
      * @returns {TrackerRule | null}
      */
-    findRule (tracker, requestData) {
+    findRule (tracker, requestData, supportedCustomRuleActions) {
         let matchedRule = null
         // Find a matching rule from this tracker
         if (tracker.rules && tracker.rules.length) {
             tracker.rules.some(ruleObj => {
-                if (this.requestMatchesRule(requestData, ruleObj)) {
+                if (this.ruleActionSupported(ruleObj, supportedCustomRuleActions) &&
```

Comment:
> 1. Recommendation: for performance reasons...

Interesting. I actually put the checks this way around for performance reasons. My thinking was that a Set lookup (or even two) for the rule action is much cheaper an operation than checking if the rule matches (checking if a regular expression matches the URL, plus more work if the rule has extra options.) But you're right, more often than not the rule action _will_ be supported.

Which is faster overall, avoiding checking if the rules match requests for some inactive Click to Load rules, or skipping the rule action Set lookups for rules that don't match a request... I'm not sure! My vote would be to leave this as is for now, but to revisit if performance turns out to be an issue WDYT?

> 2. Idea: another approach would be to just filter out unrecognized rule actions when parsing the rule set?

I don't think this is possible unfortunately. Supposing Facebook Click to Load is disabled for a tab, we _don't_ want to support the "block-ctl-fb" rule action any more for the tab's requests. I think we need the ability to specify which rule actions are active each time we check for matches.

Reply:
> 1. Recommendation: for performance reasons...

As discussed in MatterMost, I was on the fence about which was better for performance in practice without measuring it. Since you felt more strongly, I just swapped it. Done.